code/campaignclassic/src/main/java/com/adobe/marketing/mobile/campaignclassic/internal/CampaignClassicPushTrackerActivity.kt

```diff
@@ -0,0 +1,245 @@
+     */
+    private fun handlePushClicked(intent: Intent) {
+        CampaignClassic.trackNotificationClick(getTrackInfo(intent))
+        executePushAction(intent)
+    }
+
+    /**
+     * Retrieves the Campaign Classic push notification tracking information from the received
+     * notification's [Intent]
+     *
+     * @param intent the intent received from the push notification
+     * @return Map<String, String> containing the notification's tracking information
+     */
+    private fun getTrackInfo(intent: Intent): Map<String, String> {
+        val trackInfo = mutableMapOf<String, String>()
+        val extras = intent.extras ?: return trackInfo
+        for (key in extras.keySet()) {
+            val value = extras.getString(key)
+            if (value != null) {
+                if (key == CampaignClassicConstants.EventDataKeys.CampaignClassic.TRACK_INFO_KEY_MESSAGE_ID) {
+                    trackInfo[key] = value
+                } else if (key == CampaignClassicConstants.EventDataKeys.CampaignClassic.TRACK_INFO_KEY_DELIVERY_ID) {
+                    trackInfo[key] = value
+                }
+            }
+        }
+        return trackInfo
+    }
+
+    /**
+     * Reads the URI and executes the action that is configured for the push notification.
+     *
+     *
+     * If no URI is configured, by default the application will be opened.
+     *
+     * @param intent the intent received from the push notification
+     */
+    private fun executePushAction(intent: Intent) {
+        val actionUri =
+            intent.getStringExtra(PushTemplateConstants.TrackingKeys.ACTION_URI)
+        if (actionUri.isNullOrEmpty()) {
+            openApplication(intent.extras)
+        } else {
+            openUri(actionUri, intent.extras)
+        }
+
+        // remove the notification if sticky notifications are false
+        val isStickyNotification = intent.getStringExtra(PushTemplateConstants.PushPayloadKeys.STICKY)?.toBoolean() ?: false
+        val tag = intent.getStringExtra(PushTemplateConstants.PushPayloadKeys.TAG)
+        if (isStickyNotification) {
+            Log.trace(
+                CampaignClassicConstants.LOG_TAG,
+                SELF_TAG,
+                "the sticky notification setting is true, will not remove the notification" +
+                    " with tag $tag."
+            )
+            return
+        }
+        val context = ServiceProvider.getInstance().appContextService.applicationContext
+            ?: return
+        val notificationManager = NotificationManagerCompat.from(context)
+        if (tag.isNullOrEmpty()) {
+            Log.warning(
+                CampaignClassicConstants.LOG_TAG,
+                SELF_TAG,
+                "the sticky notification setting is false but the tag is null or empty," +
+                    " default to removing all displayed notifications for ${application.packageName}"
+            )
+            notificationManager.cancelAll()
```

Comment:
I am a little concerned over this logic to cancel all notification, when tag is not available.
This is anyway an edge scenario.. we wouldn't allow a notification get displayed without a tag. 

Even in this edge case, I think it would be ideal to leave the notification in the system tray rather than removing all the notifications

Reply:
I just copied existing logic 😄 It should never happen because the tag is set as the messageId if one isn't provided. @rymorale was this requirement from PM? If not, can we do nothing in this case?